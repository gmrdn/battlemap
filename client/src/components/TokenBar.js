import React from "react";
import TokenChip from "./TokenChip";
import { connect } from "react-redux";
import { addToken } from "../actions";

const TokenBar = (props) => {
  return (
    <div
      className="container-fluid overflow-auto mt-3 p-0"
      style={{ height: "50vh" }}
    >
      <h5>Token Bar</h5>
      <ul className="list-group border-0">
        {props.tokens.map((token) => {
          return (
            <li
              key={token._id}
              className="list-group-item border-0 pl-0 pr-2 pb-0 pt-0"
            >
              <TokenChip token={token} tileSide={40}></TokenChip>
            </li>
          );
        })}
      </ul>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    tokens: state.tokens,
  };
};

const mapDispatchToProps = { addToken };

export default connect(mapStateToProps, mapDispatchToProps)(TokenBar);
