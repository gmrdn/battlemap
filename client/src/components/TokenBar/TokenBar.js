import React from "react";
import TokenChip from "../TokenChip";
import { makeStyles } from "@material-ui/core/styles";
import { connect } from "react-redux";
import { openNewTokenDialog } from "../../actions";
import Box from "@material-ui/core/Box";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import { ListItemText } from "@material-ui/core";
import ExpandLess from "@material-ui/icons/ExpandLess";
import ExpandMore from "@material-ui/icons/ExpandMore";

const useStyles = makeStyles((theme) => ({
  root: {
    borderRadius: "3px",
    backgroundColor: theme.palette.background.default,
    position: "absolute",
    "max-height": "240px",
    overflowY: "auto",
    top: "200px",
    right: "0px",
  },
}));

export const TokenBar = (props) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(true);

  const handleClick = () => {
    setOpen(!open);
  };

  return (
    <Box class={classes.root}>
      <ListItem button onClick={handleClick}>
        <ListItemText primary="Tokens" />
        {open ? <ExpandLess /> : <ExpandMore />}
      </ListItem>
      <List dense>
        {props.tokens.map((token) => {
          return (
            <ListItem key={token._id}>
              <TokenChip
                token={token}
                tileSide={40}
                socket={props.socket}
              ></TokenChip>
            </ListItem>
          );
        })}
      </List>
    </Box>
  );
};

const mapStateToProps = (state) => {
  return {
    nickname: state.nickname,
    roomId: state.roomId,
    tokens: state.tokens,
  };
};

const mapDispatchToProps = { openNewTokenDialog };

export default connect(mapStateToProps, mapDispatchToProps)(TokenBar);
