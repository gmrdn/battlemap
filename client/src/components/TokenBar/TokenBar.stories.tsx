import React from "react";
import { TokenBar } from "./TokenBar";

export default {
  title: "TokenBar",
};

const socket = () => {};
const tokens = [
  {
    _id: "1234",
    x: 1,
    y: 1,
    name: "Storybook",
    color: "blue",
    image: "1.png",
    selected: false,
  },
  {
    _id: "5678",
    x: 1,
    y: 1,
    name: "Second token",
    color: "green",
    image: "12.png",
    selected: true,
  },
];

export const Default = () => (
  <TokenBar tokens={tokens} socket={socket}></TokenBar>
);

export const Empty = () => <TokenBar tokens={[]} socket={socket}></TokenBar>;
