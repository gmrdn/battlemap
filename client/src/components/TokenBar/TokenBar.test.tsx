import React from "react";
import { shallow } from "enzyme";
import { TokenBar } from "./TokenBar";

const socket = () => {};
const tokens = [
  {
    _id: "1234",
    x: 1,
    y: 1,
    name: "Storybook",
    color: "blue",
    image: "1.png",
    selected: false,
  },
  {
    _id: "5678",
    x: 1,
    y: 1,
    name: "Second token",
    color: "green",
    image: "12.png",
    selected: true,
  },
];

describe("Components", () => {
  describe("TokenBar", () => {
    it("displays a button with a text as parameter", () => {
      const wrapper = shallow(
        <TokenBar tokens={tokens} socket={socket}></TokenBar>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
