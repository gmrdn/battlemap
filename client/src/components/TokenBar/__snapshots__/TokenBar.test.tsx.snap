// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components TokenBar displays a button with a text as parameter 1`] = `
<Styled(MuiBox)
  boxShadow={3}
  class="makeStyles-root-1"
>
  <WithStyles(ForwardRef(List))
    dense={true}
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemText))
        primary="Tokens"
      />
      <ExpandLessIcon />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(Collapse))
      in={true}
      timeout="auto"
      unmountOnExit={true}
    >
      <WithStyles(ForwardRef(ListItem))
        key="1234"
      >
        <Connect(TokenChip)
          socket={[Function]}
          tileSide={40}
          token={
            Object {
              "_id": "1234",
              "color": "blue",
              "image": "1.png",
              "name": "Storybook",
              "selected": false,
              "x": 1,
              "y": 1,
            }
          }
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        key="5678"
      >
        <Connect(TokenChip)
          socket={[Function]}
          tileSide={40}
          token={
            Object {
              "_id": "5678",
              "color": "green",
              "image": "12.png",
              "name": "Second token",
              "selected": true,
              "x": 1,
              "y": 1,
            }
          }
        />
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(Collapse))>
  </WithStyles(ForwardRef(List))>
</Styled(MuiBox)>
`;
